api/v1alpha4/cluster_webhook.go

```diff
@@ -45,24 +50,36 @@ func (c *Cluster) Default() {
 	if c.Spec.ControlPlaneRef != nil && len(c.Spec.ControlPlaneRef.Namespace) == 0 {
 		c.Spec.ControlPlaneRef.Namespace = c.Namespace
 	}
+
+	// If the Cluster uses a managed topology
+	if c.Spec.Topology != nil {
+		// tolerate version strings without a "v" prefix: prepend it if it's not there
+		if !strings.HasPrefix(c.Spec.Topology.Version, "v") {
+			c.Spec.Topology.Version = "v" + c.Spec.Topology.Version
+		}
```

Comment:
Let's remove this assumption and only accept versions with `v` prefix?

Reply:
This will make this version field behave differently from the other version fields, I would prefer consistency vs being strict.

Reply:
@CecileRobertMichon thoughts?

Reply:
I'm with Fabrizio on this one, consistency is better. But we can open an issue to consider requiring the "v" prefix on all version fields across the API. Although IMO it's nice to not have to worry about it as a user, functionally it's the same.

Reply:
I'll open an issue to not make it easier for users but rather acknowledge that it should always be prefixed with `v`. The reason to drop the simplified approach is that we're making an assumption which isn't clear to the user until they get/describe the object. This can also break gitops.